presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -410,6 +407,333 @@ Drop a schema::
+
+
+``hive.domain-compaction-threshold``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:**
+
+  Maximum number of ranges/values allowed while reading hive data without compacting it.
+  A higher value will cause more data fragmentation but allow the use of the row skipping
+  feature when reading ORC data. Increasing this value may have a large impact on ``IN``
+  and ``OR`` clause performance in scenarios making use of row skipping.
+
+
+.. _force-local-scheduling:
+
+``hive.force-local-scheduling``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:**
+
+  Force splits to be scheduled on the same node (ignoring normal node selection procedures)
+  as the Hadoop DataNode process serving the split data. This is useful for installations
+  where Presto is collocated with every DataNode and may decrease queries time significantly.
+  The drawback may be that if some data are accessed more often, the utilization of some nodes
+  may be low even if the whole system is heavy loaded.
+  See also :ref:`node-scheduler.network-topology<node-scheduler-properties>` if less
+  strict constrain is preferred - especially if some nodes are overloaded and other are not
+  fully utilized.
+
+
+``hive.max-initial-split-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.max-split-size`` / ``2`` (``32 MB``)
+ * **Description:**
+
+  This property describes the maximum size of the first ``hive.max-initial-splits``
+  splits created for a query. the logic behind initial splits is described in
+  ``hive.max-initial-splits``. Lower values will increase concurrency for small queries.
+  This property represents the maximum size, as the real size may be lower when the amount
+  of data to read is less than ``hive.max-initial-split-size`` (e.g. at the end of a
+  block on a DataNode).
+
+
+``hive.max-initial-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``200``
+ * **Description:**
+
+  This property describes how many splits may be initially created for a single query
+  using ``hive.max-initial-split-size`` instead of ``hive.max-split-size``. A higher
+  value will force more splits to have a smaller size (``hive.max-initial-splits`` is
+  expected to be smaller than ``hive.max-split-size``), effectively increasing the
+  definition of what is considered a "small query". The purpose of the smaller split
+  size for the initial splits is to increase concurrency for smaller queries.
+
+
+``hive.max-outstanding-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:**
+
+  Limit on the nubmer of splits waiting to be served by a split source. After reaching
+  this limit, writers will stop writing new splits until some of hteme are used by workers.
```

Comment:
writers? does it only related to `INSERT` queries? It is a bit confusing. I thought it is about readers which will not produce any new splits in there are many waiting to be processed, am I right?

Reply:
Right. its not related to INSERTs (final writes) but to the un-scheduled splits waiting to be read.